docs/lazy-definitions.md

```diff
@@ -53,7 +40,7 @@ Map = λf λlist
 
 This code will work as expected, since `cons` and `nil` are lambdas without free variables, they will be automatically floated to new definitions if the [float-combinators](compiler-options.md#float-combinators) option is active, allowing them to be unrolled lazily by hvm.
 
-It's recommended to use a [supercombinator](https://en.wikipedia.org/wiki/Supercombinator) formulation to make terms be unrolled lazily, preventing infinite expansion in recursive function bodies.
+To handle recursion properly, recursive calls should be wrapped in top-level combinators. This ensures lazy unrolling of recursive terms, preventing infinite expansion during reduction. While [supercombinators](https://en.wikipedia.org/wiki/Supercombinator) are commonly used for this purpose, other combinator patterns can work as well, as long as they're lifted to the top level.
```

Comment:
`recursive calls should be wrapped in top-level combinators`
I don't understand what you meant by this.

The exact way that this works is a bit hard to put in words. The recursive part of the function should be part of a combinator that is not in an active position. That way it can be lifted into a top-level function which is compiled into a lazy reference thus preventing the infinite expansion.